src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -236,6 +239,9 @@ def shell_safe_json_parse(json_or_dict_string, preserve_order=False):
             return ast.literal_eval(json_or_dict_string)
         except SyntaxError:
             raise CLIError(json_ex)
+        except ValueError as ex:
+            logger.debug(ex)  # log the exception which could be a python dict parsing error.
+            raise CLIError(json_ex)  # raise json_ex error which is more readable and likely.
```

Comment:
Why is it not sufficient to do the error detection only in `shell_safe_json_parse`?

Reply:
@tjprescott, i guess i did so because get_json_file has file path information which can be added to error output, but shell_safe_json_parse does not. Moreover, some methods call get_json_file but others call shell_safe_json_parse.